### DISCUSSION TITLE
okofen update 4.04

### DISCUSSION THREAD

**Discussion by lucaferrucci** [dominikamann/oekofen-pellematic-compact#55] [Q&A] [answered]
good morning Dominik, on Saturday I upgraded the system to 4.04, from the release I saw that something has changed on the json side, in fact in homeassistant all the values ​​have become "unknown".
I'll send you the json file to see what has changed?

**Comment by lucaferrucci** [comment]
[OkoJson.txt](https://github.com/dominikamann/oekofen-pellematic-compact/files/15452481/OkoJson.txt)
[softwarechanges.pdf](https://github.com/dominikamann/oekofen-pellematic-compact/files/15452498/softwarechanges.pdf)

**Comment by dominikamann** [comment]
Hi @lucaferrucci ,

is the OkoJson.txt the result if you open the URL in webbrowser?

Can you also share the HA Error Log? 

Thanks.

**Comment by lucaferrucci** [comment]
the file is from my browser, I'll send you everything
[home-assistant_2024-05-27T13-21-38.206Z.log](https://github.com/dominikamann/oekofen-pellematic-compact/files/15456413/home-assistant_2024-05-27T13-21-38.206Z.log)

**Comment by lucaferrucci** [comment]
Questo errore ha avuto origine da un'integrazione personalizzata.

Registratore: custom_components.oekofen_pellematic_compact
Fonte: custom_components/oekofen_pellematic_compact/__init__.py:134
Integrazione: Ökofen Pellematic Compact (documentazione, problemi)
Prima occorrenza: 09:34:37 (693 occorrenze)
Ultima registrazione: 15:20:38

Error reading pellematic data
Traceback (most recent call last):
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 134, in async_refresh_api_data
    update_result = await self.fetch_pellematic_data()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 150, in fetch_pellematic_data
    result = await self._hass.async_add_executor_job(fetch_data, self._host)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 173, in fetch_data
    result = json.loads(str_response)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Invalid control character at: line 202 column 48 (char 5184)

**Comment by dominikamann** [comment]
Ok, then it's an error from Ökofen. They have added a not allowed JSON character in row 202 column 48 to there output. Can you forward this to your Ökofen friend? I can also offer them that we will test the new version before they officially release it. It seems they do not test it properly. Some easy integration tests could solve this kind of issues ...

**Reply by dominikamann** [reply]
![grafik](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/5bc913e2-2c32-48e3-9ef5-b870b2e676cb)

![grafik](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/80f1079c-1944-4c45-8d8d-8e489e68704d)

**Comment by dominikamann** [comment]
I will have a look if i can provide a workarround - that my integration will handle it ...

**Comment by lucaferrucci** [comment]
Keba takes care of the software, unfortunately I don't have direct contact with them

**Comment by lucaferrucci** [comment]
ahhh, the text you highlighted will disappear, it is a warning that when the system is restarted or reset following a software update it does not display a "ZWB" pellet quantity number but displays the text "calibration" could this be the problem?

**Comment by dominikamann** [comment, marked as answer]
Can you please try version: https://github.com/dominikamann/oekofen-pellematic-compact/releases/tag/v3.2.2. I added a workarround which might will help.

**Comment by lucaferrucci** [comment]
super domik, everything has been restored, the only incorrect value as mentioned before is the quantity of pellets which displays "calibration" instead of the numerical value "Kg", I attach my dashboard
![Okodash](https://github.com/dominikamann/oekofen-pellematic-compact/assets/110548623/63188edd-60fc-45ed-82db-061981209769)

**Comment by dominikamann** [comment]
I am pleased to hear that it is now working. The amount of pellets should work after calibration. So no need to change anything here, I suppose.

**Comment by lucaferrucci** [comment]
yess, after calibration the lalue is a number and not a text
